app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnsList/__test__/SelectColumnsList.test.tsx

```diff
@@ -0,0 +1,197 @@
+              transformationName={''}
+              selectedColumns={[]}
+            />
+          </Route>
+        </Switch>
+      </Router>
+    );
+    expect(screen.getByTestId(/select-column-list-parent/i)).toBeInTheDocument();
+  });
+  it('should render the SelectColumnsList Component with some input value along with label and null and trigger input ', () => {
+    const mockSetSelected = jest.fn();
+    render(
+      <Router history={history}>
+        <Switch>
+          <Route>
+            <SelectColumnsList
+              columnsList={[]}
+              selectedColumnsCount={0}
+              setSelectedColumns={mockSetSelected}
+              dataQuality={[
+                { label: 'hello', value: '' },
+                { label: 'world', value: '' },
+              ]}
+              transformationDataType={['all', 'test']}
+              transformationName={''}
+              selectedColumns={[]}
+            />
+          </Route>
+        </Switch>
+      </Router>
+    );
+
+    const searchIconElement = screen.getByTestId(/click-handle-focus/i);
+    fireEvent.click(searchIconElement);
+    const inputSearchElement = screen.getByTestId('input-search-id');
+    fireEvent.change(inputSearchElement, { target: { value: null } });
+    fireEvent.change(inputSearchElement, { target: { value: '123' } });
+    expect(inputSearchElement).toHaveValue('123');
+    fireEvent.change(inputSearchElement, { target: { value: 'hello' } });
+    expect(inputSearchElement).toHaveValue('hello');
+  });
+
+  it('should render the SelectColumnsList Component with some input value along with label and null and trigger searchIconClick', () => {
+    render(
+      <Router history={history}>
+        <Switch>
+          <Route>
+            <SelectColumnsList
+              columnsList={[{ label: 'hello', type: ['test'], name: 'hello' }]}
+              selectedColumnsCount={1}
+              setSelectedColumns={jest.fn()}
+              dataQuality={[
+                { label: 'hello', value: '' },
+                { label: 'world', value: '' },
+              ]}
+              transformationDataType={['test']}
+              transformationName={'join-columns'}
+              selectedColumns={[]}
+            />
+          </Route>
+        </Switch>
+      </Router>
+    );
+
+    const searchIconElement = screen.getByTestId(/click-handle-focus/i);
+    fireEvent.click(searchIconElement);
+    const inputSearchElement = screen.getByTestId('input-search-id');
+    fireEvent.change(inputSearchElement, { target: { value: null } });
+    fireEvent.change(inputSearchElement, { target: { value: '123' } });
+    expect(inputSearchElement).toHaveValue('123');
+    fireEvent.change(inputSearchElement, { target: { value: 'hello' } });
+    expect(inputSearchElement).toHaveValue('hello');
+  });
+
+  it('should render the SelectColumnsList Component while changing text input with some input value along with label and null', () => {
```

Comment:
This test just checks if an input can contain a value??

Reply:
yes, we are checking if the input will have the same value which is given.